modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/ImportTranslationResultsMVCRenderCommand.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.translation.web.internal.portlet.action;
+
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCRenderCommand;
+import com.liferay.portal.kernel.servlet.MultiSessionMessages;
+import com.liferay.translation.constants.TranslationPortletKeys;
+import com.liferay.translation.web.internal.display.context.ImportTranslationResultsDisplayContext;
+
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Alicia García
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + TranslationPortletKeys.TRANSLATION,
+		"mvc.command.name=/translation/import_translation_results"
+	},
+	service = MVCRenderCommand.class
+)
+public class ImportTranslationResultsMVCRenderCommand
+	implements MVCRenderCommand {
+
+	@Override
+	public String render(
+		RenderRequest renderRequest, RenderResponse renderResponse) {
+
+		ImportTranslationResultsDisplayContext
+			importTranslationResultsDisplayContext =
+				(ImportTranslationResultsDisplayContext)
+					MultiSessionMessages.get(
+						renderRequest,
+						ImportTranslationResultsDisplayContext.class.getName());
```

Comment:
After you make the change I suggested above (using the actual session), don't forget to *remove* the session attribute when consuming it.

Reply:
the 		
`SessionMessages.clear(renderRequest);`
do not removes the session?

I found more examples os using sessionMessages to pass attributes than with httpSession

As we talked, I will pass the paramethers, not the displayContext.